torch/csrc/jit/frontend/sugared_value.cpp

```diff
@@ -393,6 +394,14 @@ SugaredValuePtr SimpleValue::getitem(
     return std::make_shared<SimpleValue>(
         g.insert(aten::select, {val, 0, idx}, {}, loc));
   } else if (auto class_type = val_type->cast<ClassType>()) {
+    // Check if this is an indexing operation enabled by a type hint.
+    if (class_type->is_module() && type_hint) {
```

Comment:
Maybe add a comment to the extent of, we’ve already type checked this upon initial emission